lib/cocoapods/target/build_settings.rb

```diff
@@ -889,12 +900,20 @@ def pod_target_xcconfig_values_by_consumer_by_key
 
         # @return [Array<Sandbox::FileAccessor>]
         define_build_settings_method :file_accessors, :memoized => true do
-          target.file_accessors.select { |fa| fa.spec.spec_type == @xcconfig_spec_type }
+          if non_library_xcconfig?
+            target.file_accessors.select { |fa| non_library_spec == fa.spec }
+          else
+            target.file_accessors.select { |fa| fa.spec.spec_type == @xcconfig_spec_type }
+          end
         end
 
         # @return [Array<Specification::Consumer>]
         define_build_settings_method :spec_consumers, :memoized => true do
-          target.spec_consumers.select { |fa| fa.spec.spec_type == @xcconfig_spec_type }
+          if non_library_xcconfig?
```

Comment:
same question as above

Reply:
I find this way easier to read than nesting the conditional inside the `select`, personally